docs/src/catalyst_applications/bifurcation_diagrams.md

```diff
@@ -1,115 +1,118 @@
+
+This tutorial briefly introduces how to use Catalyst in conjecture with BifurcationKit through basic examples, with BifurcationKit.jl providing [a more extensive documentation](https://bifurcationkit.github.io/BifurcationKitDocs.jl/stable/). Especially for more complicated systems, where careful tuning of algorithm options might be required, reading the BifurcationKit documentation is recommended. Finally, BifurcationKit provides many additional features not described here, including [computation of periodic orbits](https://bifurcationkit.github.io/BifurcationKitDocs.jl/stable/periodicOrbit/), [tracking of bifurcation points along secondary parameters](https://bifurcationkit.github.io/BifurcationKitDocs.jl/dev/branchswitching/), and [bifurcation computations for PDEs](https://bifurcationkit.github.io/BifurcationKitDocs.jl/dev/tutorials/tutorials/#PDEs:-bifurcations-of-equilibria).
 
-First, we declare our reaction model. For this example we will use a bistable
-switch, but one which also contains a Hopf bifurcation.
+## Basic example
+For this example, we will use a modified version of the model from Wilhelm (2009)[^2] (which
+demonstrates a bistable switch as the parameter *k1* is varied). We declare the model using Catalyst:
 ```@example ex1
 using Catalyst
-rn = @reaction_network begin
-    (v0 + v*(S * X)^n / ((S*X)^n + (D*A)^n + K^n), d), ∅ ↔ X
-    (X/τ, 1/τ), ∅ ↔ A
+wilhelm_2009_model = @reaction_network begin
+    k1, Y --> 2X
+    k2, 2X --> X + Y
+    k3, X + Y --> Y
+    k4, X --> 0
+    k5, 0 --> X
 end
 ```
-Next, we specify the system parameters for which we wish to plot the bifurcation
-diagram. We also set the parameter we wish to vary in our bifurcation diagram,
-as well as the interval to vary it over. Finally, we set which variable we wish
-to plot the steady state values of in the bifurcation plot.
+Next we will create a `BifurcationProblem`. In addition to the `ReactionSystem`, we need to provide:
+- The bifurcation parameter (the parameter which is varied in the bifurcation diagram).
+- A full model parameter set. This includes the values of all non-bifurcation parameters, but also a value for the bifurcation parameter (which correspond to the point in parameter space from which the computation of the bifurcation diagram starts).
+- An initial guess of the steady state values of the system at the provided parameter set. Using this point as a start, BifurcationKit uses Newton's method to find an initial steady state from which to compute the bifurcation diagram. Hence, this guess does not need to be very exact (but may be important if the system exhibits multistability for the initial parameter set).
+- The species which concentration we wish to plot on the y-axis of the bifurcation diagram (alternatively, a custom value can be provided by using the [`record_from_solution` argument](https://bifurcationkit.github.io/BifurcationKitDocs.jl/stable/periodicOrbit/#.-record*from*solution)).
```

Comment:
```suggestion
- The species or statistic we wish to plot on the y-axis of the bifurcation diagram (a custom value can be provided by using the [`record_from_solution` argument](https://bifurcationkit.github.io/BifurcationKitDocs.jl/stable/periodicOrbit/#.-record*from*solution)).
```
Does this work with symbolic expressions? If we can intercept this to make it work with symbolic expressions that would be cool. (Maybe a good followup PR and issue?)

Reply:
Wouldn't work out of the bat. We could make a routine which takes a symbolic expression and generates a function. Long-term, I think BifurcationKit is the place for a lot of stuff like this. E.g. if it could use `NonlinearSystem`s internally it could just use all the nonlinear solvers from NonlinearSolve.jl (instead of implementing its own). It would also be possible to track the name of the bifurcation parameter and use that in the plots (instead of just calling it `p`).